### DISCUSSION TITLE
Inpainting Mask Blur

### DISCUSSION THREAD

**Discussion by MonsteredBoy** [Interpause/auto-sd-krita#42] [Q&A] [answered]
Is it just me or does this not work? I set it to 40 even, and noticed the mask was not blurred, on inverted mask or normal.

**Comment by Interpause** [comment]
Weird, I cannot reproduce the issue.

Inpainting region:

<img width="573" alt="image" src="https://user-images.githubusercontent.com/42513874/198274828-a9ff0d0e-d3a1-4307-b1f1-a2031b84d02c.png">

Mask Blur 0:

<img width="631" alt="image" src="https://user-images.githubusercontent.com/42513874/198274949-afdf3474-85d6-4ce5-ba15-4e34c1e020f0.png">

Mask Blur 128:

<img width="590" alt="image" src="https://user-images.githubusercontent.com/42513874/198275028-d26423ec-7cf7-4775-bb26-d2dee5517793.png">

Mask Blur is actually quite a subtle effect, it takes denoising strength of 1 (i.e. completely changing the inpainted region) to make it obvious.

**Reply by MonsteredBoy** [reply]
In Auto's webui, I'm able to get the effect with a denoising strength of 0.35, making it a timesaver (virtually 0 work blending the inpainted region). Will there be plans for similar functionality?

**Reply by Interpause** [reply]
You see, the thing is that I am directly calling the internal API and passing it the mask as well as the mask blur. So the fact it is not working well is actually weird. Wait let me try something.

**Comment by Interpause** [comment]
Okay, see if https://github.com/Interpause/auto-sd-krita/commit/c11ded074dc774bfb9402568a7b53184fd774f83 behaves as you expect

**Reply by MonsteredBoy** [reply]
All righty, just tried it on a clean install of your fork and I'm still seeing an obvious line even when setting mask blur to 60 in Krita. 

Admittedly, I thought I was going crazy and tried it in Auto again with mask blur 1, then mask blur 40, and there is a difference between the outputs. (I would've felt bad if I sent you on a wild goose chase, but somehow the is a difference in behaviour.)

**Reply by Interpause** [reply]
Oh, I just had a thought. Maybe its the inpaint color correction. Looking at the code, color correction is applied after the mask blurring (which would be an upstrean mistake, not mine).

If turning it off works, tell me, I'll disable color correction by default then.

**Reply by MonsteredBoy** [reply]
I turned off colour correction for both inpaint and im2img and it's still a visible line, argh. Complete settings in inpaint are: euler_a, 60 steps, CFG 11, denoising 0.35, mask blur 60 (overkill to make it obvious if it's working or not), inpaint fill on preserve, inpaint full res is on with padding at 32.

**Comment by Interpause** [comment, marked as answer]
I unarchived the repository just to mention this. Its fixed now, but the method is slightly different.
https://github.com/Interpause/auto-sd-paint-ext/wiki/Usage-Guide#inpainting
